internal/service/handlers/verify_passport.go

```diff
@@ -169,13 +191,64 @@ func fulfillPassportScanEvent(r *http.Request, balance data.Balance) error {
 		return errors.New("inconsistent state: balance has no country, event type is active, but no open event was found")
 	}
 
-	_, err = EventsQ(r).
+	event, err = EventsQ(r).
 		FilterByID(event.ID).
 		Update(data.EventFulfilled, nil, nil)
 
+	if !evTypePassport.AutoClaim || !country.ReserveAllowed || country.Reserved >= country.ReserveLimit {
+		return err
+	}
+
+	_, err = claimEventWithPoints(r, *event, evTypePassport.Reward, &balance)
 	return err
 }
 
+// if autoClaim true - events will be claimed if autoclaim enabled for event
+// if false - claim events without check if autoclaim enabled
+func claimReferralSpecificEvents(r *http.Request, balance data.Balance, autoClaim bool) error {
+	evTypeRef := EventTypes(r).Get(evtypes.TypeReferralSpecific, evtypes.FilterInactive)
+	if evTypeRef == nil {
+		Log(r).Debug("Referral specific event type is inactive")
+		return nil
+	}
+
+	if autoClaim && !evTypeRef.AutoClaim {
+		return nil
+	}
+
+	country, err := CountriesQ(r).FilterByCodes(*balance.Country).Get()
+	if err != nil {
+		return fmt.Errorf("failed to get referrer country: %w", err)
+	}
+
+	if country == nil {
+		return fmt.Errorf("failed to get referrer country: must be present in database")
```

Comment:
```suggestion
		return fmt.Errorf("critical: referrer country not found, but must be present due to previous logic")
```
you haven't failed to get it